app/models/ext_management_system.rb

```diff
@@ -226,6 +226,10 @@ def self.raw_connect?(*params)
     !!raw_connect(*params)
   end
 
+  def self.raw_event_connect?(*params)
+    !!raw_event_connect(*params)
```

Comment:
I would probably prefer to imlement ```supports :raw_event_connect```

And in the Ui we would have the if :amqp; if manager.supports?(:raw_event_connect); with some error if it doesn't support it

Reply:
that would be instead of defining this as a core method, with default true

Reply:
Yeah agreed, you can do runtime conditionals in `:supports` like [`:reboot_guest`](https://github.com/ManageIQ/manageiq-providers-vmware/blob/master/app/models/manageiq/providers/vmware/infra_manager/vm/operations/guest.rb#L5-L14) where you can do your `OpenstackRabbitEventMonitor.available?` check.

Reply:
Hmm if you need to pass params that might not work though...